java/core/src/main/java/sleeper/core/statestore/FileReferenceStore.java

```diff
@@ -26,67 +28,119 @@
     void addFiles(List<FileReference> fileReferences) throws StateStoreException;
 
-    default void splitFileReferences(List<SplitFileReferenceRequest> splitRequests) throws StateStoreException {
-    }
+    /**
+     * Performs atomic updates to split file references. This is used to push file references down the partition tree,
+     * eg. where records are ingested to a non-leaf partition, or when a partition is split. A file referenced in a
+     * larger, non-leaf partition may be split between smaller partitions which cover non-overlapping sub-ranges of the
+     * original partition. This includes these records in compactions of the descendent partitions.
+     * <p>
+     * The aim is to combine all records into a small number of files for each leaf partition, where the leaves of the
+     * partition tree should represent a separation of the data into manageable chunks. Compaction operates on file
+     * references to pull records from multiple files into one, when they are referenced in the same partition. This
+     * reduces the number of files in the system, and improves statistics and indexing within each partition. This
+     * should result in faster queries, and more accurate partitioning when a partition is split.
+     * <p>
+     * Each {@link SplitFileReferenceRequest} will remove one file reference, and create new references to the same file
+     * in descendent partitions. The ranges covered by the partitions of the new references must not overlap, so there
+     * must never be two references to the same file where one partition is a descendent of the other.
+     * <p>
+     * Note that it is possible that the necessary updates may not fit in a single transaction. Each
+     * {@link SplitFileReferenceRequest} is guaranteed to be done atomically in one transaction, but it is possible that
+     * some may succeed and some may fail. If a single {@link SplitFileReferenceRequest} adds too many references to
+     * apply in one transaction, this will also fail.
+     *
+     * @param splitRequests A list of {@link SplitFileReferenceRequest}s to apply
+     * @throws StateStoreException          if update fails
+     * @throws SplitRequestsFailedException if update fails when split into multiple transactions, and some requests may have succeeded
+     */
+    void splitFileReferences(List<SplitFileReferenceRequest> splitRequests) throws StateStoreException;
 
     /**
-     * Atomically changes the status of some files from active to ready for GC
-     * and adds new {@link FileReference}s as active files.
+     * Atomically applies the results of a job. Removes file references for a job's input files, and adds references to
+     * an output file. This will be used for compaction.
+     * <p>
+     * This should validate that the input files were assigned to the job.
+     * <p>
+     * This should result in the input files becoming available for garbage collection, if no other references exist
+     * for those files.
```

Comment:
This is an asynchronous property of the system, not a property of the state store.